### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by mikejerome**
Attempting #79 

I have never done testing so I used the docs, google, and @TheAlphaNerd's submission as a guide. This was a good learning experience.

### CODE REVIEW

**Comment by mikejerome** [comment]
Why my Travis failed?

**Comment by TheAlphaNerd** [comment]
If you click on the red x you can see.  It actually looks like it might have been a system error on travis.  I noticed the last commit is a "merge" commit.  You probably want to rebase that out anywho... so if you clean up that commit and force push you should re run travis

**Comment by mikejerome** [comment]
I have not done a rebase before please advise.

**Comment by hussainanjar** [comment]
@mikejerome read this guide https://github.com/hueniverse/hueniversity/blob/master/guides/git.md to see how you can rebase

**Comment by mikejerome** [comment]
@hussainanjar that guide doesn't seem to mention rebasing. I have been using a separate branch for my changes.

**Comment by TheAlphaNerd** [comment]
@mikejerome here is an article on squashing commits with rebase

http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by TheAlphaNerd** [comment]
I highly suggest making a new branch to experiment with first though

**Comment by mikejerome** [comment]
Awesome, thanks @TheAlphaNerd

File: lib/index.js [hunk 1]
```diff
@@ -10,19 +9,24 @@ var Version = require('./version');
 var internals = {};
 
 
-internals.init = function () {
+exports.init = function (port, callback) {
+
+    if (typeof port === 'function') {
```
Comment (hueniverse) [inline lib/index.js]: Not in assignment.

**Pull request closed without merging**